src/scripts/carousel.js

```diff
@@ -0,0 +1,129 @@
+(function() {
+  "use strict";
+
+  // Variables
+
+  var pBox = document.querySelectorAll(".carousel-box");
+  var pCarousel = document.querySelector(".products-carousel");
+  var carouselDots = document.querySelector(".carousel-dots");
+  var dotsMain = document.querySelector(".dots-main");
+  var boxArray = [];
+
+  // Dynamically added dots that depend on the number of product boxes
+
+  function addDots() {
+    var box;
+    var vPortWidth = window.screen.availWidth;
+    var pBoxL = pBox.length;
+    var i;
+    var dotsTempl = '<li><span class="dot-click basic-animation"></span></li>';
+    carouselDots.innerHTML = "";
+
+    if (vPortWidth >= 1025) {
```

Comment:
w if else wystarczy że dasz obliczanie box a poza ifką dasz tą pętle i returna bowiem wykonujesz ją zawsze

Reply:
A to święta racja. Tak czułem że tu coś można zmienić. Co zdrowy człowiek to nie chory.... ;)